src/fsharp/FSharp.Core/iseq.fs

```diff
@@ -0,0 +1,1971 @@
+                    let finalIndex  = match lastOption with
+                                        | Some b -> b             // here b>=0, a valid end value.
+                                        | None   -> unreachable   // run "forever", well as far as Int32.MaxValue since indexing with a bounded type.
+                    // The Current value for a valid index is "f i".
+                    // Lazy<_> values are used as caches, to store either the result or an exception if thrown.
+                    // These "Lazy<_>" caches are created only on the first call to current and forced immediately.
+                    // The lazy creation of the cache nodes means enumerations that skip many Current values are not delayed by GC.
+                    // For example, the full enumeration of Seq.initInfinite in the tests.
+                    // state
+                    let index   = ref unstarted
+                    // a Lazy node to cache the result/exception
+                    let current = ref (Unchecked.defaultof<_>)
+                    let setIndex i = index := i; current := (Unchecked.defaultof<_>) // cache node unprimed, initialised on demand.
+                    let getCurrent() =
+                        if !index = unstarted then notStarted()
+                        if !index = completed then alreadyFinished()
+                        match box !current with
+                        | null -> current := Lazy<_>.Create(fun () -> f !index)
+                        | _ ->  ()
+                        // forced or re-forced immediately.
+                        (!current).Force()
+                    { new IEnumerator<'U> with
+                            member x.Current = getCurrent()
+                        interface IEnumerator with
+                            member x.Current = box (getCurrent())
+                            member x.MoveNext() =
+                                if !index = completed then
+                                    false
+                                elif !index = unstarted then
+                                    setIndex 0
+                                    true
+                                else (
+                                    if !index = System.Int32.MaxValue then raise <| System.InvalidOperationException (SR.GetString(SR.enumerationPastIntMaxValue))
+                                    if !index = finalIndex then
+                                        false
+                                    else
+                                        setIndex (!index + 1)
+                                        true
+                                )
+                            member this.Reset() = noReset()
+                        interface System.IDisposable with
+                            member x.Dispose () = () }
+
+            type EnumerableDecider<'T>(count:Nullable<int>, f:int->'T, pipeIdx:PipeIdx) =
```

Comment:
`///` please.  When I read this first time I had no idea what a ``Decider`` is. It seems to be only used for ``init``?